### ISSUE TITLE
[Enterprise Bot Template][C#] Dependency Injection not used for GraphClient

### ISSUE THREAD

**Issue by hansmbakker** [microsoft/botframework-solutions#527]
**Project**
Enterprise Bot Template

**Description**
Dependency Injection is not used for `GraphClient`, instead a new instance is created every time at https://github.com/Microsoft/AI/blob/master/templates/Enterprise-Template/src/csharp/EnterpriseBotSample/Dialogs/Authentication/AuthenticationDialog.cs#L72 and in https://github.com/Microsoft/AI/blob/master/templates/Enterprise-Template/src/csharp/EnterpriseBotSample/ServiceClients/GraphClient.cs

Is this intentional?

**Comment by lauren-mills** [comment]
Hi @hansmbakker, this is intentional in the template, because AuthenticationDialog is a sample dialog and the graph client is not required for all scenarios. This should be more clear in the docs, so I've added a work item to the Enterprise Bot project board.

In a production bot, this should be initialized in startup.cs.

**Comment by hansmbakker** [comment]
@lauren-mills thank you for the clarification!

If it should be registered in `startup.cs`, how is it supposed to be consumed (injected) in a `Dialog` then?

cc @drub0y @cleemullins @johnataylor

I’m asking this since the current practice seems to be creating all services as a singleton property of `BotServices` which is passed to the constructors of all `Dialog`s using the `new` keyword (looking at the Enterprise Bot template and assuming that Microsoft put best practices in there).

Also please see https://github.com/microsoft/botbuilder-dotnet/issues/1188.

Bottom line: it is unclear how to properly use dependency injection in Bot Framework `Dialog`s and if it is possible at all?
